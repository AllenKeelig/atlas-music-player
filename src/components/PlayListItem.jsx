// src/components/PlaylistItem.jsx
export default function PlaylistItem({ title, artist, duration, isActive }) {
  return (
    <div
      className={`flex items-center justify-between p-3 rounded-lg cursor-pointer ${
        isActive ? "bg-blue-600 text-white" : "bg-gray-100 hover:bg-gray-200"
      }`}
    >
      {/* Left side: Song title + artist */}
      <div>
        <p className="font-medium">{title}</p>
        <p className="text-sm text-gray-500">{artist}</p>
      </div>

      {/* Right side: Duration */}
      <span className="text-sm">{duration}</span>
    </div>
  );
}
