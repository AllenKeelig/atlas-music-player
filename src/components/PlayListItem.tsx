type PlayListItemProps = {
  title: string;
  artist: string;
  duration: string;
  isActive?: boolean;
};

export default function PlayListItem({
  title,
  artist,
  duration,
  isActive = false,
}: PlayListItemProps) {
  return (
    <div
      className={`flex items-center justify-between p-3 rounded-lg cursor-pointer ${
        isActive
          ? "bg-secondary text-background"
          : "bg-background text-primary hover:bg-accent"
      }`}
    >
      <div>
        <p className="font-medium">{title}</p>
        <p className="text-sm text-accent">{artist}</p>
      </div>
      <span className="text-sm">{duration}</span>
    </div>
  );
}
