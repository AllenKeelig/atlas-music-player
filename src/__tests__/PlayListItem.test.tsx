// src/__tests__/PlayListItem.test.tsx
import { render } from "@testing-library/react";
import { expect, test } from "vitest";
import PlayListItem from "../components/PlayListItem";

test("PlayListItem renders selected", () => {
  const { container } = render(
    <PlayListItem
      id="1"
      title="Test Song"
      artist="Test Artist"
      isSelected={true}
      onClick={() => {}}
    />
  );
  expect(container).toMatchSnapshot();
});

test("PlayListItem renders unselected", () => {
  const { container } = render(
    <PlayListItem
      id="2"
      title="Another Song"
      artist="Another Artist"
      isSelected={false}
      onClick={() => {}}
    />
  );
  expect(container).toMatchSnapshot();
});
