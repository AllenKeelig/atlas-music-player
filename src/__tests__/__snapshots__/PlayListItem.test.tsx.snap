// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PlayListItem renders selected 1`] = `
<div>
  <div
    class="flex items-center justify-between p-3 rounded-lg cursor-pointer bg-background text-primary hover:bg-accent "
    role="button"
  >
    <div>
      <p
        class="font-medium"
      >
        Test Song
      </p>
      <p
        class="text-sm text-accent"
      >
        Test Artist
      </p>
    </div>
    <span
      class="text-sm"
    />
  </div>
</div>
`;

exports[`PlayListItem renders unselected 1`] = `
<div>
  <div
    class="flex items-center justify-between p-3 rounded-lg cursor-pointer bg-background text-primary hover:bg-accent "
    role="button"
  >
    <div>
      <p
        class="font-medium"
      >
        Another Song
      </p>
      <p
        class="text-sm text-accent"
      >
        Another Artist
      </p>
    </div>
    <span
      class="text-sm"
    />
  </div>
</div>
`;
